src/routes/UserSettings/EditProfileTab/EditProfileTab.js

```diff
@@ -10,31 +10,124 @@ import Loading from 'components/Loading'
-export function linkedinPrompt (prompt) {
-  var url = window.prompt(prompt || 'Please enter the full url for your LinkedIn page.')
-  if (url) {
-    if (validateLinkedin(url)) {
-      return url
-    } else {
-      return linkedinPrompt('Invalid url. Please enter full url for your LinkedIn page.')
+/** LinkedIn Url */
+const validateLinkedinUrl = url => url.match(/^(http(s)?:\/\/)?([\w]+\.)?linkedin\.com/)
+
+export const linkedinPrompt = () => {
+  let linkedinUrl = window.prompt('Please enter the full url for your LinkedIn page.')
+
+  if (linkedinUrl) {
+    while (!validateLinkedinUrl(linkedinUrl)) {
+      linkedinUrl = window.prompt('Invalid url. Please enter the full url for your LinkedIn page.')
     }
   }
+
+  return linkedinUrl
 }
 
-export const validateLinkedin = url => url.match(/^(http(s)?:\/\/)?([\w]+\.)?linkedin\.com/)
+/** Facebook Url */
+const validateFacebookUrl = url => url.match(/^(http(s)?:\/\/)?([\w]+\.)?facebook\.com/)
+
+export const facebookPrompt = () => {
+  let facebookUrl = window.prompt('Please enter the full url for your Facebook page.')
+
+  if (facebookUrl) {
+    while (!validateFacebookUrl(facebookUrl)) {
+      facebookUrl = window.prompt('Invalid url. Please enter the full url for your Facebook page.')
+    }
+  }
+
+  return facebookUrl
+}
+export class SocialControl extends Component {
+  static propTypes = {
+    label: string,
+    provider: string,
+    value: string | null,
+    updateSettingDirectly: func,
+    handleUnlinkAccount: func,
+    onLink: func
+  }
+
+  handleLinkClick () {
+    const { provider, updateSettingDirectly } = this.props
+
+    switch (provider) {
+      case 'twitter': {
+        const twitterHandle = window.prompt('Please enter your twitter name.')
+        if (twitterHandle) {
+          updateSettingDirectly()(twitterHandle)
+        }
+        break
+      }
+      case 'linkedin': {
+        const linkedinUrl = linkedinPrompt()
+        if (linkedinUrl) {
+          updateSettingDirectly()(linkedinUrl)
+        }
+        break
+      }
+      case 'facebook': {
+        const facebookUrl = facebookPrompt()
+        if (facebookUrl) {
+          updateSettingDirectly()(facebookUrl)
+        }
+        break
+      }
+    }
+  }
+
+  handleUnlinkClick () {
+    const { handleUnlinkAccount, updateSettingDirectly } = this.props
+
+    handleUnlinkAccount()
+    updateSettingDirectly()(null)
+  }
+
+  render () {
+    const { label, value = '', provider, onLink } = this.props
+    const linked = !!value
+
+    const linkButton =
+      <span
+        styleName='link-button'
+        onClick={linked ? () => this.handleUnlinkClick() : () => this.handleLinkClick()}>
+        {linked ? 'Unlink' : 'Link'}
+
+      </span>
 
+    const connectFacebookButton =
+      <span
+        styleName='link-button'
+        onClick={linked ? () => this.handleUnlinkClick() : () => onLink()}
+        className='ml-auto'
+      >
+        {linked ? 'Disconnect' : 'Connect'}
+      </span>
```

Comment:
I think the only outstanding thing on this PR is whether or not this connect facebook flow works :/  I'm just not sure how to test it :s

Reply:
that can be left as a separate issue since i dont know how to test locally right now either :(